middle_end/flambda2/terms/bound_symbols.ml

```diff
@@ -120,7 +128,27 @@ type t = Pattern.t list
 
 let empty = []
 
-let create pattern_list = pattern_list
+let check_pattern_list_invariant pattern_list =
+  (* Check that there are no repeated bindings of symbols or code IDs. *)
+  let everything_being_defined =
+    List.map Pattern.everything_being_defined_as_list pattern_list
+    |> List.concat
+  in
+  let everything_being_defined_as_set =
+    Code_id_or_symbol.Set.of_list everything_being_defined
+  in
+  if List.compare_length_with everything_being_defined
+       (Code_id_or_symbol.Set.cardinal everything_being_defined_as_set)
+     <> 0
+  then
+    Misc.fatal_errorf "Illegal pattern list:@ %a"
```

Comment:
Maybe you could specify here why the pattern list is illegal ? Namely that there is a repeated symbol being bound.

Reply:
Fixed